app/src/main/java/org/jellyfin/androidtv/ui/search/SearchViewModel.kt

```diff
@@ -0,0 +1,70 @@
+package org.jellyfin.androidtv.ui.search
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.util.apiclient.waitForSuccess
+import org.jellyfin.sdk.model.api.BaseItemKind
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+
+class SearchViewModel : ViewModel() {
+
+	private var searchJob: Job? = null
+
+	private var previousQuery: String? = null
+
+	private val _searchResultsFlow = MutableStateFlow<List<SearchResultGroup>>(emptyList())
+	val searchResultsFlow = _searchResultsFlow.asStateFlow()
+
+	fun search(query: String) = searchDebounced(query, 0.milliseconds)
```

Comment:
Maybe call this one "searchImmediatly" so it's more obvious what it does

Reply:
Changed